src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -68,6 +68,26 @@ def validate_keyvault(cmd, namespace):
                                           'vaults', 'Microsoft.KeyVault')
 
 
+def validate_proximity_placement_group(cmd, namespace):
+    from msrestazure.tools import parse_resource_id, is_valid_resource_id
+
+    namespace.proximity_placement_group = _get_resource_id(cmd.cli_ctx, namespace.proximity_placement_group,
+                                                           namespace.resource_group_name,
+                                                           'proximityPlacementGroups', 'Microsoft.Compute')
+
+    if not namespace.proximity_placement_group or not is_valid_resource_id(namespace.proximity_placement_group):
+        logger.debug('Proximity placement group is %s', namespace.proximity_placement_group)
+        return
+
+    parsed = parse_resource_id(namespace.proximity_placement_group)
+
+    rg = parsed['resource_group']
+    name = parsed['name']
+
+    if not check_existence(cmd.cli_ctx, name, rg, 'Microsoft.Compute', 'proximityPlacementGroups'):
```

Comment:
what will happen at service end, if we don't do the check at client end? Do we get a good error?

Reply:
I believe the error should be something like the following, if the ppg is in a different region or does not exist:

```
Deployment failed. Correlation ID: 260fc599-b6f4-418f-9c2e-fc800602d78f. {
  "error": {
    "code": "NotFound",
    "message": "Proximity Placement Group '/subscriptions/00977cdb-163f-435f-9c32-39ec8ae61f4d/resourceGroups/ova-test/providers/Microsoft.Compute/proximityPlacementGroups/tosinppg' cannot be found."
  }
}
```

I can remove this existence check, however the CLI does perform this check for availability set...